WooCommerce/Classes/ViewRelated/Products/ShareProduct/ShareProductAIEligibilityChecker.swift

```diff
@@ -0,0 +1,16 @@
+import protocol Experiments.FeatureFlagService
+import struct Yosemite.Site
+
+struct ShareProductAIEligibilityChecker {
+    private let site: Site?
+    private let featureFlagService: FeatureFlagService
+
+    init(site: Site?, featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
+        self.site = site
+        self.featureFlagService = featureFlagService
+    }
+
+    var canGenerateShareProductMessageUsingAI: Bool {
+        site?.isWordPressComStore == true && featureFlagService.isFeatureFlagEnabled(.shareProductAI)
```

Comment:
the product form also [checks if the site is public](https://github.com/woocommerce/woocommerce-ios/blob/1fe491770e5e75a567c9646eee0d64d06e4cdd5c/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/ProductFormViewModel.swift#L264) to determine if a product can be shared, maybe if it's already checked somewhere else?

Reply:
Thanks for sharing this, Jaclyn. 

1. The "Share product" option will be visible only if the site is public. i.e. The AI eligibility check will be invoked only for public sites. 
1. Also, I think that a site being public is not really a requirement for `ShareProductAIEligibilityChecker`. (AFAIK, AI content can be generated as long as the site is WPCOM) 

Kindly let me know if you think it will be good to check for `site.isPublic` here.

Reply:
if it's already a condition for `canGenerateShareProductMessageUsingAI` to be called, no need to check again 👍 (this is the first PR I've reviewed on the share product feature so I was just wondering)

Reply:
Sure, I understand. Thanks for your feedback. 🙏